### PR TITLE
Pagerduty destination doc

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

> Describe the changes you are proposing in this PR.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

**Review by mihir-4116 — APPROVED** [review]

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 1]
```diff
@@ -0,0 +1,172 @@
+---
+title: "PagerDuty"
+description: Send your event data from RudderStack to PagerDuty.
+aliases: ["PagerDuty destination"]
+---
+
+[PagerDuty](https://www.pagerduty.com/) is a SaaS-based platform that helps prevent and resolve business-impacting incidents, thereby improving the customer experience.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:7]:
```suggestion
[PagerDuty](https://www.pagerduty.com/) is a SaaS platform that helps you prevent and resolve business-impacting incidents, thereby improving the customer experience.
```

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+<div class="infoBlock">
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:9]:
```suggestion
```

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/pagerduty.mdx:9]: Reducing the no. of callouts as suggested by Pete.

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/pagerduty">GitHub repository</a>.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:10]:
```suggestion
Find the open source code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/pagerduty">GitHub repository</a>.
```

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+</div>
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:11]:
```suggestion
```

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+| `originalTimeStamp`     | `payload.timestamp`      | 
+| `properties.customDetails`   | `payload.custom_details`     |
+
+## FAQ
+
+### Where can I find the PagerDuty Routing Key?
+
+1. Log in to the [PagerDuty dashboard](https://app.pagerduty.com/).
+2. Go to **Services** > **Service Directory**.
+3. Select your desired service and click **Integrations** tab.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:169]:
```suggestion
3. Select your desired service and click the **Integrations** tab.
```

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+4. Click **Add integration** and select any desired integration platform. Here, you can find the integration/routing key as shown:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:170]:
```suggestion
4. Click **Add integration** and select the integration platform. Here, you can find the integration/routing key as shown:
```

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+Once you have confirmed that the source platform supports sending events to PagerDuty, follow these steps:
+
+1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **PagerDuty**.
+2. Assign a name to the destination and click **Continue**.
+
+## Connection settings
+
+To successfully configure PagerDuty as a destination, you will need to configure the following settings:
+
+- **PagerDuty Routing Key**: Enter the integration key from PagerDuty service. Refer to the <a href="#faq">FAQ</a> section for more information on how to obtain it.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:31]:
```suggestion
- **PagerDuty Routing Key**: Enter the integration key associated with your PagerDuty service. Refer to the <a href="#faq">FAQ</a> section for more information on how to obtain it.
```

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/pagerduty.mdx:31]: LMK if this change makes sense.

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+- **dedup_key Field Identifier**: Enter a `dedup_key` which acts as an identifier for the event. You can provide any value in this field. If not provided, PagerDuty automatically generates and assigns a `dedup_key` to the event. This key is used while updating, resolving, or acknowledging an event.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:32]:
```suggestion
- **dedup_key Field Identifier**: Enter a `dedup_key` which acts as the event identifier and is used while updating, resolving, or acknowledging an event. You can provide any value in this field. If not provided, PagerDuty automatically generates and assigns a `dedup_key` to the event.
```

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+## Track
+
+You can send a <Link to="/event-spec/standard-events/track">`track`</Link> call to PagerDuty to ingest the [Alert events](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event) or [Change events](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgy-send-a-change-event).
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:36]: Slightly confused with the PagerDuty docs - they refer to `Alert events` in lower case (`alert events` as seen [here](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event#:~:text=Every-,alert%20event,-has%20a%20dedup_key), but `Change Event` is written in title case. Anyway, I think the way you have presented it looks the best.

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+### Alert events
+
+You can [send an alert event](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event) to report a new problem, or update an ongoing problem, depending on the event type.
+
+A sample `track` call for sending an alert event is shown below:
+
+```javascript
+rudderanalytics.track("apiSecret is not present", {
+  action: "trigger",
+  dedupKey: "9116b734-7e6b-4497-ab51-c16744d4487e",
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:57]: In the connection settings, we write this as `dedup_key` (snake case) while here it's in camel case. Please confirm if we need to update the web app, @mihir-4116 cc @ishachopra0401

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+  severity: "critical",
+  component: "ui",
+  source: "rudder-webapp",
+  group: "destination",
+  class: "connection settings",
+  customDetails: {
+    "ping time": "1500ms",
+    "load avg": 0.75,
+  },
+  imageURLs: [{
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:67]: @mihir-4116 do we need all these URLs? for a sample snippet, just 1 should be fine?

File: docs/destinations/streaming-destinations/pagerduty.mdx [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+    imageURLs: [{
+        src: "https://static.s4be.cochrane.org/app/uploads/2017/04/shutterstock_531145954.jpg",
+        alt: "first image",
+      },
+      {
+        src: "https://chart.googleapis.com/chart?chs=600x400&chd=t:6,2,9,5,2,5,7,4,8,2,1&cht=lc&chds=a&chxt=y&chm=D,0033FF,0,0,5,1",
+        alt: "second image",
+      },
+    ],
+    linkURLs: [{
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/pagerduty.mdx:131]: Same observation as above, we can mention just one image and link URL for this sample snippet.

**Comment by ameypv-rudder** [comment]
Resolved some of the comments after discussing with @mihir-4116.

**Pull request merged**